src/main/java/xyz/holocons/mc/holoitemsrevamp/enchantment/Splinter.java

```diff
@@ -0,0 +1,151 @@
+    }
+
+    @Override
+    public int getCostMultiplier() {
+        return 12;
+    }
+
+    @Override
+    public void onBlockBreak(BlockBreakEvent event, ItemStack itemStack) {
+        Block blockBroken = event.getBlock();
+        if(isInvalidSplinterType(blockBroken.getType()) || isInvalidSplinterLocation(blockBroken.getLocation())){
+            return;
+        }
+        // Presuming the block is a log - ie, it has an orientation.
+        tryCleanupSplinterData(event.getPlayer(), blockBroken);
+
+        // Check and handle if the block is a trunk
+        // TODO
+
+        // Check and handle if the block is a branch
+        // TODO
+    }
+
+    /**
+     * Try to cleanup a player's Splinter data. If the data is scheduled for cleanup, but the newTrunk isn't a valid
+     * new trunk block, it will delete the data from the map entirely.
+     * @param player The player's
+     * @param newTrunk The proposed new trunk block
+     */
+    private void tryCleanupSplinterData(Player player, Block newTrunk){
+        final var activeSplinter = splinters.get(player);
+        // if(!shouldCleanup)
+        if(activeSplinter != null && activeSplinter.scheduledSplinters != 0){
+            // there is still a splinter going, so don't cleanup until that one's done
+            return;
+        }
+
+        splinters.remove(player);
+
+        BlockData data = newTrunk.getBlockData();
+        if(!(data instanceof Orientable)){
+            // all logs are orientable, so this isn't a log
+            // TODO: ... mushrooms?
+            return;
+        }
+
+        Axis orientationAxis = ((Orientable) data).getAxis();
+        if(orientationAxis != Axis.Y){
+            return;
+        }
+
+        ActiveSplinter newData = new ActiveSplinter();
+        newData.origin = newTrunk.getLocation().clone();
+        newData.remainingCharges = 32; // In the future we can make this based on enchLevel
+        newData.scheduledSplinters = 0;
+        splinters.put(player, newData);
+    }
+
+    private void handleTrunkBroken(ActiveSplinter data, Block trunkBlock){
+        // TODO
+    }
+
+    private void handleBranchBroken(ActiveSplinter data, Block branchBlock){
+        // TODO
+    }
+
+    private void scheduleSplinterAbility(Player player, Block block) {
+        // TODO
+    }
+
+    private boolean isInvalidSplinterType(Material type) {
+        return !this.COMPATIBLE_MATERIALS.isTagged(type);
+    }
+
+    private boolean isInvalidSplinterLocation(Location loc){
+        return !Integrations.WORLDGUARD.canUseEnchantment(loc, Splinter.class);
+    }
+
+    private static boolean isCompatibleMaterial(Material material) {
+        return Tag.LOGS.isTagged(material) || Tag.CRIMSON_STEMS.isTagged(material)
+            || Tag.WARPED_STEMS.isTagged(material) || Tag.WART_BLOCKS.isTagged(material)
+            || MaterialTags.MUSHROOM_BLOCKS.isTagged(material);
+    }
+
+    private static class ActiveSplinter {
+        Location origin;
```

Comment:
```suggestion
        Block origin;
```
Seems like we might want to store the block since we could get blockX, blockY, blockZ, and type from it.

Reply:
Yep. Had Location instead of Block because I couldn't think of any reason we might need the block's type, but I forgot we don't want splinter to change block-types (aka from spruce logs to oak logs).

Reply:
Aside from using the type, we probably also want to use the block positions. When you use Location, the positions get converted to floats.